src/attr/validators.py

```diff
@@ -592,3 +593,41 @@ def min_len(length):
     .. versionadded:: 22.1.0
     """
     return _MinLengthValidator(length)
+
+
+@attrs(repr=False, slots=True, hash=True)
+class _NotValidator:
+    validator = attrib()
+
+    def __call__(self, inst, attr_, value):
+        try:
+            self.validator(inst, attr_, value)
+        except ValueError:
+            pass  # suppress error to invert validity
+        else:
+            message = "not_ validator child '{validator}' did not raise error"
+            raise ValueError(message.format(validator=repr(self.validator)))
+
+    def __repr__(self):
+        return "<not_ validator wrapping {what}>".format(
+            what=repr(self.validator)
+        )
+
+
+def not_(validator):
+    """
+    A validator that wraps and logically 'inverts' the validator passed to it,
```

Comment:
This needs to be expanded to TypeErrors. Sorry that it disturbs the nice flow of your sentence. :)

Reply:
Also idle though: this could be made configurable if the need ever arises. As in you pass in exactly the exception you want to require.